### PR TITLE
Adding recipe for knem-headers

### PR DESCRIPTION
**Pull request by j34ni**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [X] Source is from official source.
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [X] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [X] Build number is 0.
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by j34ni** [comment]
The knem project comprises multiple components, including a user-space library and a crucial kernel module that facilitates efficient memory copy operations at the kernel level. This package specifically provides the knem_io.h header from the knem project, allowing other software, like UCX, to interface with the knem kernel module, enabling high-performance communication in distributed computing environments.

File: recipes/knem-headers/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    - {{ pin_subpackage('knem-headers', max_pin='x.y.z') }}
+
+requirements:
+  build:
+  host:
+  run:
+
+test:
+  commands:
+    - ls -la $PREFIX/include/knem_io.h
```
Comment (xhochy) [inline recipes/knem-headers/meta.yaml:26]:
```suggestion
    - test -f $PREFIX/include/knem_io.h
```

File: recipes/knem-headers/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+source:
+  url: https://gitlab.inria.fr/knem/knem/-/archive/knem-{{ version }}/knem-knem-{{ version }}.tar.gz
+  sha256: 2196777458626c467057dd7eaee915154acda47dcd0600edb271ef7535198e7b
+
+build:
+  number: {{ build }}
+  skip: true  # [win or osx]
+
+  run_exports:
+    - {{ pin_subpackage('knem-headers', max_pin='x.y.z') }}
```
Comment (xhochy) [inline recipes/knem-headers/meta.yaml:17]: If these are only headers, there should be no runtime dependency.

File: recipes/knem-headers/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+package:
+  name: knem-headers
+  version: {{ version }}
+
+source:
+  url: https://gitlab.inria.fr/knem/knem/-/archive/knem-{{ version }}/knem-knem-{{ version }}.tar.gz
+  sha256: 2196777458626c467057dd7eaee915154acda47dcd0600edb271ef7535198e7b
+
+build:
+  number: {{ build }}
```
Comment (xhochy) [inline recipes/knem-headers/meta.yaml:13]:
```suggestion
  number: 0
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/knem-headers/meta.yaml (hunk 2 above)
Comment (j34ni) [reply recipes/knem-headers/meta.yaml:17]: OK

**Comment by j34ni** [comment]
@xhochy Is that OK now?

**Review by xhochy — APPROVED** [review]

**Pull request merged**